packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -108,10 +108,9 @@ class FlutterManifest {
   /// Can be null if version isn't set or has a wrong format.
   String get buildNumber {
     if (appVersion != null && appVersion.contains('+')) {
-      final String value = appVersion.split('+')?.elementAt(1);
-      return value;
+      return appVersion.split('+')?.elementAt(1);
     } else {
-      return null;
+      return appVersion;
```

Comment:
So, if the app version is "1.0+1", then this function returns "1", but if it's "1.0", then it returns "1.0"?

Seems like you want to return null here, not the `appVersion`.